Metric/MetricHandler.php

```diff
@@ -175,7 +175,7 @@ public function getFormattedMetric(MetricInterface $metric)
             '%tags' => $this->formatTagsInline(
                 $metric->getResolvedTags([
                     'container' => $this->container,
-                    'request' => $this->requestStack ? $this->requestStack->getMasterRequest() : null,
+                    'request' => $this->requestStack ? $this->requestStack->getCurrentRequest() : null,
```

Comment:
I think you should use MasterRequest in default case anyway.

Reply:
KernelEvents are not guaranteed to bear the master request, should I ignore those and keep the request fetched here from the stack ?

Reply:
Yes, I think you should keep the MasterRequest - fetched here from the stack